src/ncp/ncp_base.cpp

```diff
@@ -3161,6 +3164,64 @@ ThreadError NcpBase::SetPropertyHandler_THREAD_CONTEXT_REUSE_DELAY(uint8_t heade
     return errorCode;
 }
 
+#if OPENTHREAD_ENABLE_DIAG
+ThreadError NcpBase::SetPropertyHandler_NEST_STREAM_MFG(uint8_t header, spinel_prop_key_t key, const uint8_t *value_ptr, uint16_t value_len)
+{
+    char *string(NULL);
+    char *output(NULL);
+    spinel_ssize_t parsedLength;
+    ThreadError errorCode = kThreadError_None;
+
+    parsedLength = spinel_datatype_unpack(
+                       value_ptr,
+                       value_len,
+                       SPINEL_DATATYPE_UTF8_S,
+                       &string
+                   );
+
+    if ((parsedLength > 0) && (string != NULL))
+    {
+        char *argv[8];
+        int argc = 0;
+        int length = static_cast<int>(strlen(string));
+        char *cmd;
+
+        for (; *string == ' '; string++, length--);
+
+        argv[argc++] = string;
+
+        for (cmd = string + 1; (cmd < string + length) && (cmd != NULL); ++cmd)
+        {
+            if (*cmd == ' ' || *cmd == '\r' || *cmd == '\n')
+            {
+                *cmd = '\0';
+            }
+
+            if (*(cmd - 1) == '\0' && *cmd != ' ')
+            {
+                argv[argc++] = cmd;
+            }
+        }
```

Comment:
I feel like this line parsing logic should be handled outside of the NCP. To the NCP, the contents of this string should be completely opaque. There should be a function called something like `diagProcessCmdLine()` that just takes a single string argument, or perhaps a function for converting a command string into `argc`/`argv` (less preferable).